perllib/FixMyStreet/DB/ResultSet/DefectType.pm

```diff
@@ -3,20 +3,38 @@ use base 'DBIx::Class::ResultSet';
 
 use strict;
 use warnings;
+use JSON::MaybeXS;
 
 sub for_bodies {
     my ($rs, $bodies, $category) = @_;
     my $attrs = {
         'me.body_id' => $bodies,
     };
-    if ($category) {
-        $attrs->{'contact.category'} = [ $category, undef ];
-    }
-    $rs->search($attrs, {
+    my $filters = {
         order_by => 'name',
         join => { 'contact_defect_types' => 'contact' },
         distinct => 1,
-    });
+    };
+    if ($category) {
+        $attrs->{'contact.category'} = [ $category, undef ];
+    }
+    $rs->search($attrs, $filters);
+}
+
+sub by_categories {
+    my ($rs, $area_id, @contacts) = @_;
+    my $body_id = FixMyStreet::DB->resultset('BodyArea')->find({ area_id => $area_id })->body_id;
```

Comment:
I've got the code to do this, but when writing tests and trying to create the extra body like so:

```
my $area_id = 123;
my $other_body = $mech->create_body_ok($area_id, 'Some Other Council');
FixMyStreet::App->model('DB::BodyArea')->create( { body => $other_body, area_id => $area_id } );
```

I get an error from Postgres:

```
DBIx::Class::ResultSet::create(): DBI Exception: DBD::Pg::st execute failed: ERROR:  duplicate key value violates unique constraint "body_areas_body_id_area_id_idx"
DETAIL:  Key (body_id, area_id)=(1, 123) already exists. [for Statement "INSERT INTO body_areas ( area_id, body_id) VALUES ( ?, ? )" with ParamValues: 1='123', 2='1'] at t/app/model/responsepriority.t line 73
    # Child (by_categories returns defect types if an area is covered by more than one body) exited without calling finalize()
```

This suggests that there's a constraint that stops bodies having multiple areas, so this will never happen. Am I missing something?